src/task1.1/reverser.ts

```diff
@@ -0,0 +1,33 @@
+import { Transform, TransformOptions, TransformCallback } from 'stream';
+
+export class Reverser extends Transform {
+  constructor(options?: TransformOptions) {
+    super(options);
+    this.on('pipe', () => {
+      console.log('Please enter the string and I will reverse it.');
+    });
+    this.on('error', (err) => {
+      console.error(err.message);
+    });
+  }
+  prepareChunk(chunk: Buffer): string {
+    return chunk.toString('utf-8').replace('\n', '');
+  }
+  processData(data: string): string {
+    return data.split('').reverse().join('').concat('\n\n');
+  }
+  _transform(
+    chunk: Buffer,
+    enc: BufferEncoding,
+    callback: TransformCallback
+  ): void {
+    const data = this.prepareChunk(chunk);
+    if (!data) {
+      callback(null, 'Error: The input is empty \n');
```

Comment:
Sorry, my bad - concat is not used in this case, but you have <Buffer 0d 0a> in case of empty line.
So you need to replace "\r\n" instead of "\n" in the "prepareChunk" method to make condition working.

Reply:
Actually it was working without it. toString transformed Buffer to empty string. But I have updated regexp a bit